File: misc/ses_sendmail_lambda.py
import json
import boto3

from botocore.exceptions import ClientError

def lambda_handler(event, context):

    SENDER = 'alanli8519@gmail.com'
    RECIPIENT = 'alanli8519@gmail.com'
    SUBJECT = 'TEST'
    BODY_TEXT = 'TEST BODY'
    BODY_HTML = ''
    CHARSET = "UTF-8"
    AWS_REGION = 'us-west-2'
    
    SESclient = boto3.client('ses',region_name=AWS_REGION)
    
    try:
        resp = SESclient.send_email(
            Destination={
                'ToAddresses': [
                    RECIPIENT,
                ],
            },
            Message={
                'Body': {
                    'Html': {
                        'Charset': CHARSET,
                        'Data': BODY_HTML,
                    },
                    'Text': {
                        'Charset': CHARSET,
                        'Data': BODY_TEXT,
                    },
                },
                'Subject': {
                    'Charset': CHARSET,
                    'Data': SUBJECT,
                },
            },
            Source=SENDER,
        )
    except ClientError as e:
        return {
            'statusCode' : 400,
            'body' : json.dumps(e)
        }
    else:
        return {
            'statusCode' : 200,
            'body' : json.dumps('SUCCESS')
        }

File: misc/ses_sendmail_lambda.js
import { SESClient, SendEmailCommand } from "@aws-sdk/client-ses"; // ES Modules import
// const { SESClient, SendEmailCommand } = require("@aws-sdk/client-ses"); // CommonJS import

export const handler = async(event) => {
    const CHARSET = 'UTF-8'
    let   sender;
    let   recipient;
    let   content_text;
    let   content_html;
    let   subject;

    const SESClient = new SESClient({
        region: 'us-west-2'
    });

    const sendMailInput = {
        Source: sender,
        Destination: {ToAddresses: [recipient]},
        Message: { 
            Subject: { 
                Data: subject,
                Charset: CHARSET
            },
            Body: { 
                Text: {
                    Data: content_text,
                    Charset: CHARSET
                },
                Html: {
                    Data: content_html,
                    Charset: CHARSET
                },
            },
        },
    }

    try{
        const sendEmail = new SendEmailCommand(sendMailInput);
        const sendMailResp = await SESClient.send(sendEmail);
        console.log(sendMailResp);

        return{
            statuscode: 200,
            body: "SUCCESS"
        }
    }catch(err){
        return{
            statuscode: 400,
            body: err
        }
    }
}